dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/JSONFieldsTest.java

```diff
@@ -0,0 +1,128 @@
+    }
+
+    public void setUp(LibertyServer server) throws Exception {
+        if (server != null && !server.isStarted()) {
+            // Restore the original server configuration, before starting the server for each test case.
+            server.restoreServerConfiguration();
+            server.startServer();
+        }
+    }
+
+//    @After
+    public void cleanUp(LibertyServer server) throws Exception {
+        if (server != null && server.isStarted()) {
+            server.stopServer("com.ibm.ws.logging.fat.ffdc.servlet.FFDCServlet.doGet", "ArithmeticException",
+                              "CWWKG0081E", "CWWKG0083W");
+        }
+    }
+
+    /*
+     * This test sets the "WLP_LOGGING_JSON_FIELDS" attribute in the server.env and verifies the property in the messages.log file.
+     */
+    @Test
+    public void testMessageFieldNamesEnv() throws Exception {
+        setUp(server_env);
+        List<String> lines = server_env.findStringsInFileInLibertyServerRoot("log3", MESSAGE_LOG);
+        assertTrue("The message field name was not formatted in the new configuration in messages.log.", lines.size() > 0);
+    }
+
+    /*
+     * This test sets the "com.ibm.ws.logging.json.fields" attribute in the bootstrap.properties and verifies the property in the messages.log file.
+     */
+    @Test
+    public void testMessageFieldNamesBootstrap() throws Exception {
+        setUp(server_bootstrap);
+        List<String> lines = server_bootstrap.findStringsInFileInLibertyServerRoot("log2", MESSAGE_LOG);
+        assertTrue("The message field name was not formatted in the new configuration in messages.log.", lines.size() > 0);
+    }
+
+    /*
+     * This test sets the "jsonFields" attribute in the server.xml and verifies the property in the messages.log file.
+     */
+    @Test
+    public void testMessageFieldNamesXML() throws Exception {
+        setUp(server_xml);
+        setServerConfiguration(true, "message:test1", server_xml);
+
+        List<String> lines = server_xml.findStringsInFileInLibertyServerRoot("test1", MESSAGE_LOG);
+        assertTrue("The message field name was not formatted in the new configuration in messages.log.", lines.size() > 0);
+        setServerConfiguration(true, "message:test2", server_xml);
+        lines = server_xml.findStringsInFileInLibertyServerRoot("test2", MESSAGE_LOG);
+        assertTrue("The message field name was not updated in messages.log.", lines.size() > 0);
+    }
+
+    @Test
+    public void jsonFieldsErrorChecking() throws Exception {
+        // Set jsonFields property in server.xml
+        setUp(server_xml);
+        //map the ibm_datetime field to blank field name
+        setServerConfiguration(true, "ibm_datetime:", server_xml);
+
+        //the default field name should be used
+        List<String> lines = server_xml.findStringsInFileInLibertyServerRoot("ibm_datetime", MESSAGE_LOG);
+        assertTrue("The default field name was not returned", lines.size() > 0);
+    }
+
```

Comment:
need more tests for badly formed jsonField strings.  eg. for field names that don't match ours, or new field names that have spaces or other bad JSON characters, or wrong number of tokens, etc.